File: its-my-training-day/algorithms/leetcode-top-75/questions/arrays-hashing/hash-map/contains-duplicate/solution.ts
import {
  ArrayList,
  Stack,
  Queue,
  ListNode,
  TreeNode,
  MinHeap,
  MaxHeap,
} from "../../../../shared/typescript/ds";

export const containsDuplicate = (nums: number[]): boolean => {
  const seen: Record<number, boolean> = {};
  for (let i = 0; i < nums.length; i += 1) {
    if (seen[nums[i]]) return true;
    seen[nums[i]] = true;
  }
  return false;
};

File: its-my-training-day/algorithms/leetcode-top-75/questions/arrays-hashing/hash-map/contains-duplicate/solution.py
import os
import sys
from typing import List, Dict, Optional, Tuple

ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../../"))
if ROOT_DIR not in sys.path:
    sys.path.append(ROOT_DIR)

from shared.python.ds import ArrayList, ListNode, Stack, Queue, TreeNode, MinHeap, MaxHeap

def contains_duplicate(nums: List[int]) -> bool:
    seen: Dict[int, bool] = {}
    i = 0
    while i < len(nums):
        val = nums[i]
        if val in seen:
            return True
        seen[val] = True
        i += 1
    return False
